src/components/table/column-picker/column-picker.component.html

```diff
@@ -12,39 +12,86 @@
         </ng-container>
     </div>
 
-    <div class="column-picker-list" [(uxSelection)]="_deselectedSelection">
+    <cdk-tree class="column-picker-list"
+              [dataSource]="_treeDataSource"
+              [treeControl]="_treeControl"
+              [(uxSelection)]="_deselectedSelection"
+              uxTabbableList
+              role="tree">
+
+        <!-- Create item for not expandable node -->
+        <cdk-tree-node *cdkTreeNodeDef="let node">
+            <div uxTabbableListItem
+                 [uxSelectionItem]="node"
+                 *ngIf="_shouldRenderNode(node)"
+                 role="treeitem"
+                 class="column-picker-list-item column-picker-tree-node-level-{{ node.level }}">
+
+                <ng-container *ngIf="!deselectedTemplate">{{ node.name }}</ng-container>
+
+                <ng-container
+                    *ngIf="deselectedTemplate"
+                    [ngTemplateOutlet]="deselectedTemplate"
+                    [ngTemplateOutletContext]="{ $implicit: node.name }">
+                </ng-container>
+            </div>
 
-        <div *ngFor="let column of deselected"
-             class="column-picker-list-item"
-             [uxSelectionItem]="column">
+        </cdk-tree-node>
 
-            <ng-container *ngIf="!deselectedTemplate">{{ column }}</ng-container>
+        <!-- Create item for expandable node -->
+        <cdk-tree-node *cdkTreeNodeDef="let node; when: _nodeHasChildren" role="treeitem">
 
-            <ng-container
-                *ngIf="deselectedTemplate"
-                [ngTemplateOutlet]="deselectedTemplate"
-                [ngTemplateOutletContext]="{ $implicit: column }">
-            </ng-container>
-        </div>
-    </div>
+            <div *ngIf="_nodeHasAvailableChildren(node)"
+                class="column-picker-tree-group-node">
+
+                <button uxTabbableListItem
+                        (click)="_setNodeExpanded(node, !node.isExpanded)"
+                        (keydown.arrowright)="_setNodeExpanded(node, true)"
+                        (keydown.arrowleft)="_setNodeExpanded(node, false)"
+                        [style.visibility]="node.expandable ? 'visible' : 'hidden'"
+                        [attr.aria-label]="getDeselectedAriaLabel(node)"
```

Comment:
Just had another look at this and it's not quite right. It should be:
```
[attr.aria-label]="deselectedAriaLabel(node)"
```
Otherwise any custom function provided by the application will be ignored. The `getDeselectedAriaLabel` function should be renamed to `getDefaultDeselectedAriaLabel` and made private.

Reply:
Good point that's a mistake on my behalf. I've followed the same pattern for the selectedAriaLabel to keep them consistent.

Reply:
Yes, looks like that one is wrong as well, but we can fix that another time.